### DISCUSSION TITLE
VPN Access

### DISCUSSION THREAD

**Discussion by metaskills** [community/community#5767] [Codespaces] [answered]
Is this possible? I have a `postCreateCommand` hook that does the following to install open connect.

```
sudo apt-get update
sudo apt-get -y install software-properties-common
sudo add-apt-repository -y ppa:phusen/openconnect || true # for universal
sudo apt-get -y install openconnect xmlstarlet
```

However, I am getting the following error. Even when run as root or with sudo. I've only tried this on the `mcr.microsoft.com/vscode/devcontainers/ruby:2.5` image so not done a lot of debugging yet.

```
> Failed to bind local tun device (TUNSETIFF): Operation not permitted
To configure local networking, openconnect must be running as root
See http://www.infradead.org/openconnect/nonroot.html for more information
Set up tun device failed
Unknown error; exiting.
```

**Comment by metaskills** [comment, marked as answer]
OK, I found out this does work. There are two things. First, the `universal` image has a small error when running `add-apt-repository` but it can be ignored. Updated script above with a true fallback. The second and most important is that you need to include these in your `devcontainer.json` file to fix the tun error above.

```json
{
  "runArgs": ["--init", "--privileged"]
}
```

**Comment by metaskills** [comment]
Another point... if you are using `dockerComposeFile` then `runArgs` will not work. Instead use this setting in your main service.

```yaml
my_service:
  privileged: true
```

**Comment by Amirhosseinyahyayi** [comment]
I need a vpn free
